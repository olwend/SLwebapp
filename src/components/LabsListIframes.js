import React, { useState, useEffect } from 'react'
import { Tab } from 'react-bootstrap'
import Iframe from 'react-iframe'

import { useSelector } from 'react-redux'
import { getSession } from '../redux/slices/sessionSlice'

import NoDataIndication from './NoDataIndication'

import { populateTabs, labsFetchData } from '../utils/iframes'

import './LabsListIframes.css'

const LabsListIframes = () => {

    const sessionInfo = useSelector(getSession)

    const [tabs, setTabs] = useState(populateTabs(sessionInfo))

    let isDelayedRefreshAlreadyActivated = false
    let currentRefreshNumber = 1

    useEffect(() => {
        labsFetchData(sessionInfo, tabs, setTabs, isDelayedRefreshAlreadyActivated, currentRefreshNumber) // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    return (
    <>
        { tabs.map(_ => {
            return(
                <Tab.Pane key={_.shortName + ':' + _.port + '-' + _.key} className="full-size-no-padding" eventKey={_.shortName + ':' + _.port}>
                    { sessionInfo.isLabRunning && _.available ? <Iframe id="theia" className="full-size-no-padding" url={_.url}/> : <div className={'no-data-indication'}><NoDataIndication /></div> }
                </Tab.Pane>
            )
        })}
    </>
    )
}

export default LabsListIframes
