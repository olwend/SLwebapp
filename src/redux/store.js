import { configureStore } from '@reduxjs/toolkit';

import userReducer from './slices/userSlice';
import sessionReducer from './slices/sessionSlice';
import labsReducer from './slices/labsSlice';

export default configureStore({
  reducer: {
    user: userReducer,
    session: sessionReducer,
    labs: labsReducer,
  },
});
