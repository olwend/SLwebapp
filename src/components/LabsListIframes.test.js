import React from 'react'

import { shallow } from 'enzyme'

import store from '../redux/store'
import { Provider } from 'react-redux'

import LabsListIframes from './LabsListIframes'

test('Renders properly', () => expect(shallow(<Provider store={store}><LabsListIframes /></Provider>).dive()).toMatchSnapshot())
