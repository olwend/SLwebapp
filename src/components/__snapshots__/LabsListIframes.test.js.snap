// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `<LabsListIframes />`;
